tests/tools/s3/utils.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+import json
+import os
+import re
+import sys
+
+import boto3
+import botocore.client
+
+
+def resource_has_tags(resource, tags={}):
+    """Checks if a resource defines all `tags`."""
+    return all(
+        tag_key in resource.keys() and resource[tag_key] == tag_value
+        for (tag_key, tag_value) in tags.items()
+    )
+
+
+class S3ResourceFetcher:
+    """A class for fetching vmm-reference test resources from S3."""
+
+    def __init__(
+            self,
+            resource_bucket,
+            resource_manifest_path,
+            download_location=None,
+    ):
+        """Initialize S3 client, manifest of test resources and S3 bucket name."""
+        with open(resource_manifest_path) as json_file:
+            self._resource_manifest = json.load(json_file)
+        self._latest_version = self.get_latest_version()
```

Comment:
The only place this is used is below inside `download` the `or clause can directly use it, this attribute doesn't become a required one.